### PR TITLE
Clean up and change docker container

### PR DESCRIPTION
**Pull request by PayamEmami**
This PR hopefully fixes the issues here: https://github.com/nf-core/metaboigniter/pull/18
The main file has been cleaned up and we now use a single docker container.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metaboigniter/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/metaboigniter _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by PayamEmami** [comment]
Forgot to write that we also have a full test for AWS

File: main.nf [hunk 1]
```diff
@@ -555,7 +595,7 @@ if(params.type_of_ionization in (["pos","both"]))
 
       process process_ipo_param_pos_ipo_centroided{
           label 'ipo'
-          tag "$name"
+          tag "A collection of files"
```
Comment (ewels) [inline main.nf:598]: If not dynamic then no need for a `tag` I think - better to just remove it.

```suggestion
```

File: main.nf [hunk 2]
```diff
@@ -568,24 +608,67 @@ if(params.type_of_ionization in (["pos","both"]))
...
-             factorGap_h=!{params.ipo_factorGap_h_pos} localAlignment=!{params.ipo_localAlignment_pos} ipo_gapInit_l=!{params.ipo_gapInit_l_pos} ipo_gapInit_h=!{params.ipo_gapInit_h_pos} ipo_gapExtend_l=!{params.ipo_gapExtend_l_pos} \
-             ipo_gapExtend_h=!{params.ipo_gapExtend_h_pos} ipo_profStep_l=!{params.ipo_profStep_l_pos} ipo_profStep_h=!{params.ipo_profStep_h_pos} bw_l=!{params.ipo_bw_l_pos} bw_h=!{params.ipo_bw_h_pos} minfrac_l=!{params.ipo_minfrac_l_pos} \
-              minfrac_h=!{params.ipo_minfrac_h_pos} mzwid_l=!{params.ipo_mzwid_l_pos} mzwid_h=!{params.ipo_mzwid_h_pos} minsamp_l=!{params.ipo_minsamp_l_pos} \
-               minsamp_h=!{params.ipo_minsamp_h_pos} max_l=!{params.ipo_max_l_pos} max_h=!{params.ipo_max_h_pos} ncores=!{params.ipo_ncores_pos} outputxset=quant_params_pos.json outputrt=rt_params_pos.json
+	 /usr/local/bin/ipo.r \\
+		input=$inputs_aggregated \\
+		quantOnly=$ipo_pos_globalAvoidRT \\
+		allSamples=$params.ipo_allSamples_pos \\
+		columnToSelect=$params.ipo_columnToSelect_pos  \\
+	 valueToSelect=$params.ipo_valueToSelect_pos \\
```
Comment (ewels) [inline main.nf:620]: Is this varied indentation intentional?

File: main.nf [hunk 3]
```diff
@@ -690,7 +821,7 @@ if(params.type_of_ionization in (["pos","both"]))
 
     process process_ipo_param_pos_ipo_noncentroided{
         label 'ipo'
-        tag "$name"
+        tag "A collection of files"
```
Comment (ewels) [inline main.nf:824]:
```suggestion
```

File: main.nf [hunk 4]
```diff
@@ -1332,7 +1669,7 @@ process  process_ms2_identification_pos_metfrag{
 
 process  process_identification_aggregate_pos_metfrag{
   label 'msnbase'
-  tag "$name"
+  tag "A collection of files"
```
Comment (ewels) [inline main.nf:1672]:
```suggestion
```

File: main.nf [hunk 5]
```diff
@@ -776,7 +982,7 @@ prefilter_h_l=!{params.ipo_prefilter_h_l_pos} prefilter_h_h=!{params.ipo_prefilt
 
   process  process_collect_rdata_pos_xcms{
     label 'xcms'
-    tag "$name"
+    tag "A collection of files"
```
Comment (ewels) [inline main.nf:985]:
```suggestion
```

File: main.nf [hunk 6]
```diff
@@ -1082,7 +1357,7 @@ if(params.perform_identification==true)
 
   process  process_mapmsms_tocamera_pos_msnbase{
     label 'msnbase'
-    tag "$name"
+    tag "A collection of files"
```
Comment (ewels) [inline main.nf:1360]:
```suggestion
```

File: main.nf [hunk 7]
```diff
@@ -1159,26 +1447,33 @@ process  process_ms2_identification_pos_csifingerid{
...
+    """
 }
 /*
  * STEP 19 - aggregate ids from CSI
  */
 
 process  process_identification_aggregate_pos_csifingerid{
   label 'msnbase'
-  tag "$name"
+  tag "A collection of files"
```
Comment (ewels) [inline main.nf:1476]:
```suggestion
```

File: main.nf [hunk 8]
```diff
@@ -1482,7 +1865,7 @@ process  process_ms2_identification_pos_cfmid{
 
 process  process_identification_aggregate_pos_cfmid{
   label 'msnbase'
-  tag "$name"
+  tag "A collection of files"
```
Comment (ewels) [inline main.nf:1868]:
```suggestion
```

File: main.nf [hunk 9]
```diff
@@ -1675,7 +2108,7 @@ if(params.library_charactrized_pos==false){
 
       process process_ipo_param_library_pos_ipo_centroided{
           label 'ipo'
-          tag "$name"
+          tag "A collection of files"
```
Comment (ewels) [inline main.nf:2111]:
```suggestion
```

File: main.nf [hunk 10]
```diff
@@ -2084,7 +2691,7 @@ mapmsmstocamera_rdata_library_pos_camerams2=ch1mapmsmsLibrary_pos.join(ch2mapmsm
 
   process  process_collect_library_pos_msnbase{
     label 'msnbase'
-    tag "$name"
+    tag "A collection of files"
```
Comment (ewels) [inline main.nf:2694]:
```suggestion
```

File: main.nf [hunk 11]
```diff
@@ -2109,7 +2719,7 @@ mapmsmstocamera_rdata_library_pos_camerams2=ch1mapmsmsLibrary_pos.join(ch2mapmsm
 
 process process_remove_adducts_library_pos_msnbase{
   label 'msnbase'
-  tag "$name"
+  tag "A collection of files"
```
Comment (ewels) [inline main.nf:2722]:
```suggestion
```

File: main.nf [hunk 12]
```diff
@@ -2393,7 +3073,7 @@ if(params.type_of_ionization in (["neg","both"]))
 
       process process_ipo_param_neg_ipo_centroided{
           label 'ipo'
-          tag "$name"
+          tag "A collection of files"
```
Comment (ewels) [inline main.nf:3076]:
```suggestion
```

File: main.nf [hunk 13]
```diff
@@ -2529,7 +3298,7 @@ if(params.type_of_ionization in (["neg","both"]))
 
     process process_ipo_param_neg_ipo_noncentroided{
         label 'ipo'
-        tag "$name"
+        tag "A collection of files"
```
Comment (ewels) [inline main.nf:3301]:
```suggestion
```

**Review by ewels — APPROVED** [review]
Great stuff! A few issues but I think it's better to just merge this PR and continue the review in https://github.com/nf-core/metaboigniter/pull/18

* Static strings for a `tag` don't really help anything, so I think these are better deleted
* You didn't need to delete the `shell:` statements - in fact we usually say that they should be there, even if Nextflow works without them. Its presence is not linked to the use of `"""` or `'''`
* Whitespace is all over the place still

**Pull request merged**